tests/fast-integration/monitoring/grafana.js

```diff
@@ -15,75 +14,74 @@ const {
+
 
 async function assertPodsExist() {
-  const namespace = 'kyma-system';
-  await waitForPodWithLabel('app', 'grafana', namespace);
+  await waitForPodWithLabel('app', 'grafana', kymaNs);
 }
 
 async function assertGrafanaRedirectsExist() {
   if (getEnvOrDefault('KYMA_MAJOR_VERSION', '2') === '2') {
-    await checkGrafanaRedirectsInKyma2();
+    await assertGrafanaRedirectsInKyma2();
   } else {
-    await checkGrafanaRedirectsInKyma1();
+    await assertGrafanaRedirectsInKyma1();
   }
 }
 
-async function checkGrafanaRedirectsInKyma2() {
-  // Checking grafana redirect to kyma docs
-  let res = await assertGrafanaRedirect('https://kyma-project.io/docs');
+async function assertGrafanaRedirectsInKyma2() {
+  info('Checking grafana redirect for kyma docs');
+  let res = await checkGrafanaRedirect('https://kyma-project.io/docs', 403);
   assert.isTrue(res, 'Grafana redirect to kyma docs does not work!');
 
-  // Creating secret for auth proxy redirect
   await manageSecret('create');
   await restartProxyPod();
-  // Checking grafana redirect to OIDC provider
-  res = await assertGrafanaRedirect('https://accounts.google.com/signin/oauth');
+
+  info('Checking grafana redirect for google as OIDC provider');
+  res = await checkGrafanaRedirect('https://accounts.google.com/signin/oauth', 200);
   assert.isTrue(res, 'Grafana redirect to google does not work!');
 
   await updateProxyDeployment('--reverse-proxy=true', '--trusted-ip=0.0.0.0/0');
-  // Checking that authentication works and redirects to grafana URL
-  res = await assertGrafanaRedirect('https://grafana.');
-  assert.isTrue(res, 'Grafana redirect to grafana landing page does not work!');
 
-  res = await resetProxy();
-  assert.isTrue(res, 'Grafana Authproxy is not reset successfully!  ');
+  info('Checking grafana redirect to grafana URL');
+  res = await checkGrafanaRedirect('https://grafana.', 200);
+  assert.isTrue(res, 'Grafana redirect to grafana landing page does not work!');
 }
 
-async function checkGrafanaRedirectsInKyma1() {
-  const res = await assertGrafanaRedirect('https://dex.');
+async function assertGrafanaRedirectsInKyma1() {
+  info('Checking grafana redirect for dex');
+  const res = await checkGrafanaRedirect('https://dex.', 200);
   assert.isTrue(res, 'Grafana redirect to dex does not work!');
 }
 
-async function assertGrafanaRedirect(redirectURL) {
-  const vs = await getVirtualService('kyma-system', 'monitoring-grafana');
-  const hosts = vs.spec.hosts[0];
-  let ignoreSSL = false;
-  if (hosts.includes('local.kyma.dev')) {
-    ignoreSSL = true;
-  }
-  const url = 'https://' + hosts;
-  if (redirectURL.includes('https://dex.')) {
-    console.log('Checking redirect for dex');
-    return await retryUrl(url, redirectURL, ignoreSSL, 200);
-  }
-
-  if (redirectURL.includes('https://kyma-project.io/docs')) {
-    console.log('Checking redirect for kyma docs');
-    return await retryUrl(url, redirectURL, ignoreSSL, 403);
+async function setGrafanaProxy() {
```

Comment:
IMHO, the function name does not describe its purpose. Maybe something like `exposeGrafana`\`unexposeGrafana`?